File: Service1/Service1/Controllers/StatusController.cs
using Microsoft.AspNetCore.Mvc;
using Service1.Services.Contracts;

namespace Service1.Controllers;

[ApiController]
public class StatusController : ControllerBase
{
    private readonly IStatusService _statusService;
    
    public StatusController(IStatusService statusService)
    {
        _statusService = statusService;
    }

    [HttpGet]
    [Route("status")]
    public IActionResult GetStatus()
    {
        var status = _statusService.GetStatus();
        return Ok(status);
    }
}   

File: Service2/src/main/kotlin/com/exercise01/service2/controllers/StatusController.kt
package com.exercise01.service2.controllers

import com.exercise01.service2.services.contracts.IStatusService
import com.exercise01.service2.services.contracts.IStorageService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class StatusController(
    val statusService: IStatusService,
    val storageService: IStorageService
) {
    @GetMapping(path = ["/status"], produces = [MediaType.TEXT_PLAIN_VALUE])
    fun getStatus(): String {
        val status = statusService.getStatus()
        storageService.addLog(status)
        return status
    }
}